src/dialect/mod.rs

```diff
@@ -918,6 +918,16 @@ pub trait Dialect: Debug + Any {
     fn supports_array_typedef_size(&self) -> bool {
         false
     }
+
+    /// Returns true if the dialect supports `ORDER BY ALL`.
+    /// `ALL` which means all columns of the SELECT clause.
+    ///
+    /// DuckDB: <https://duckdb.org/docs/sql/query_syntax/orderby.html#order-by-all>
+    /// ClickHouse: <https://clickhouse.com/docs/en/sql-reference/statements/select/order-by>
```

Comment:
```suggestion
```
might not need to list the dialects since they're already listed in the impls. That way this list doesnt go out of sync if future dialects are added